<launch>

  <arg name="device_ip" default="172.16.107.201"/>
  <arg name="frequency" value="5.0"/>
  <arg name="frame_id" default="velodyne"/>
  <arg name="publish_cloud" default="true"/>

  <include file="$(find velodyne_puck_driver)/launch/velodyne_puck_driver_nodelet.launch">
    <arg name="device_ip" value="$(arg device_ip)"/>
    <arg name="frame_id" value="$(arg frame_id)"/>
  </include>

  <group ns="$(arg frame_id)">
    <node pkg="nodelet" type="nodelet" name="velodyne_puck_decoder_nodelet" args="load velodyne_puck_decoder/VelodynePuckDecoderNodelet
    velodyne_puck_nodelet_manager" output="screen">
      <param name="frame_id" value="$(arg frame_id)"/>

      <param name="min_range" value="0.3"/>
      <param name="max_range" value="100.0"/>
      <param name="frequency" type="double" value="$(arg frequency)"/>
      <param name="publish_cloud" type="bool" value="$(arg publish_cloud)"/>

      <remap from="~cloud" to="cloud"/>
      <remap from="~packet" to="packet"/>
      <remap from="~sweep" to="sweep"/>
    </node>
  </group>

</launch>
